tesseract_ros/tesseract_rviz/src/render_tools/manipulation_widget.cpp

```diff
@@ -271,27 +271,28 @@ bool ManipulationWidget::changeManipulator(const QString& manipulator)
 
     const auto& scene_graph = tesseract_->getEnvironmentConst()->getSceneGraph();
     std::vector<std::string> joint_names = inv_kin_->getJointNames();
+    const Eigen::MatrixX2d& limits = inv_kin_->getLimits();
     inv_seed_.resize(inv_kin_->numJoints());
     int i = 0;
     joint_values_property_->removeChildren();
     for (auto& j : joint_names)
     {
-      inv_seed_[i++] = joints_[j];
+      inv_seed_[i] = joints_[j];
       const auto& joint = scene_graph->getJoint(j);
-      QString joint_description = "Limits: [inf, inf]";
-      if (joint->limits)
-        joint_description = QString("Limits: [%1, %2]")
-                                .arg(QString("%1").arg(joint->limits->lower), QString("%1").arg(joint->limits->upper));
+      QString joint_description =
+          QString("Limits: [%1, %2]").arg(QString("%1").arg(limits(i, 0)), QString("%1").arg(limits(i, 1)));
 
       rviz::FloatProperty* joint_value_property = new rviz::FloatProperty(QString::fromStdString(j),
                                                                           static_cast<float>(joints_[j]),
                                                                           joint_description,
                                                                           nullptr,
                                                                           SLOT(userInputJointValuesChanged()),
                                                                           this);
-      joint_value_property->setMin(static_cast<float>(joint->limits->lower));
-      joint_value_property->setMax(static_cast<float>(joint->limits->upper));
+
+      joint_value_property->setMin(static_cast<float>(limits(i, 0)));
```

Comment:
This is necessary for continuous joints somehow?

Reply:
I thought it was easier to pull from the kinematics instead which has already parsed the joint limits and evaluated the correct limits based on the joint type. Instead I would need to check the joint type here.